src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/SpellCheck/ICSharpSpellCheckService.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.CodeAnalysis.Razor.SpellCheck;
+
+internal interface ICSharpSpellCheckService
```

Comment:
I am trying to find out more details on the design and responsibilities of `ICSharpSpellCheckService` and `ISpellCheckService`. Can you please share if there is more background that I'm missing?

Reply:
Just want to make sure I fully understand the context and implications of these changes. It looks like `ISpellCheckService` centralizes the spell check logic and `ICSharpSpellCheckService` handles C# specific spell checking. But specifically, at a high level how do they interact with other components like document contexts and mapping services?